components/Vote/ElectionCandidacy.js

```diff
@@ -0,0 +1,540 @@
+      const {updateCandidacy, me} = this.props
+      const { values } = this.state
+
+      this.setState({updating: true})
+
+      return updateCandidacy({
+        slug: ELECTION_SLUG,
+        statement: values.statement,
+        credential: values.credential,
+        disclosures: values.disclosures,
+        birthday: values.birthday && values.birthday.length
+          ? values.birthday.trim()
+          : null,
+        portrait: values.portraitPreview ? values.portrait : undefined,
+        address: {
+          name: me.name,
+          line1: values.line1,
+          line2: values.line2,
+          postalCode: values.postalCode,
+          city: values.city,
+          country: values.country
+        }
+      }).then(() => {
+        return new Promise(resolve => setTimeout(resolve, 400))
+      }).then(() => {
+        this.setState(() => ({
+          isEditing: false,
+          updating: false,
+          error: null
+        }))
+      }).then(() => window.scrollTo(0, 0))
+        .catch((error) => {
+          this.setState(() => ({
+            updating: false,
+            error
+          }))
+        })
+    }
+
+    this.cancel = async () => {
+      const {cancelCandidacy} = this.props
+      cancelCandidacy(ELECTION_SLUG).then(() => {
+        this.setState(() => ({
+          isEditing: false,
+          error: null
+        }))
+      }).then(() => window.scrollTo(0, 0))
+        .catch((error) => {
+          this.setState(() => ({
+            updating: false,
+            error
+          }))
+        })
+    }
+
+    this.onChange = fields => {
+      this.setState(FieldSet.utils.mergeFields(fields))
+    }
+  }
+
+  deriveStateFromProps ({data}) {
+    const {statement, birthday, disclosures, credentials, address, portrait} = data.me || {}
+    const {line1, line2, city, postalCode, country = DEFAULT_COUNTRY} = address || {}
+    const credential = credentials ? credentials.find(c => c.isListed) : {}
+    return {
+      values: {
+        portrait,
+        statement,
+        birthday,
+        disclosures,
+        credentials,
+        line1,
+        line2,
+        city,
+        postalCode,
+        country,
+        credential: credential ? credential.description : undefined
+      }
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (nextProps.data.me && nextProps.data.me !== this.props.data.me) {
+      this.setState(this.deriveStateFromProps(nextProps))
+    }
+  }
+
+  render () {
+    const { values, errors, error, dirty, isEditing, updating } = this.state
+    const {url, t, vt} = this.props
+    const { data } = this.props
+    const {me = {}, election = {}} = data
+
+    const meta = {
+      title: `${vt('info/title')}: ${vt('info/candidacy/title')}`,
+      description: vt('info/description')
+    }
+
+    const candidate = !updating && election.candidates &&
+      election.candidates.find(c => c.user.id === me.id)
+
+    const isValid = !Object.values(errors).some(Boolean)
```

Comment:
you'll need to add a polyfill (lib/polyfill) for IE11 or switch to `Object.keys`